iceoryx_posh/test/moduletests/test_popo_active_call_set.cpp

```diff
@@ -0,0 +1,986 @@
+// limitations under the License.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+
+#include "iceoryx_posh/iceoryx_posh_types.hpp"
+#include "iceoryx_posh/internal/popo/building_blocks/condition_variable_data.hpp"
+#include "iceoryx_posh/popo/active_call_set.hpp"
+#include "iceoryx_posh/popo/user_trigger.hpp"
+#include "iceoryx_utils/cxx/optional.hpp"
+#include "iceoryx_utils/cxx/vector.hpp"
+#include "iceoryx_utils/internal/concurrent/smart_lock.hpp"
+#include "iceoryx_utils/posix_wrapper/semaphore.hpp"
+#include "mocks/wait_set_mock.hpp"
+#include "test.hpp"
+#include "testutils/timing_test.hpp"
+#include "testutils/watch_dog.hpp"
+
+#include <chrono>
+#include <memory>
+#include <thread>
+
+using namespace ::testing;
+using ::testing::Return;
+using namespace iox::popo;
+using namespace iox::cxx;
+using namespace iox::units::duration_literals;
+
+namespace
+{
+enum SimpleEvent
+{
+    StoepselBachelorParty,
+    Hypnotoad
+};
+
+uint64_t g_invalidateTriggerId = 0U;
+class SimpleEventClass
+{
+  public:
+    SimpleEventClass() = default;
+    SimpleEventClass(const SimpleEventClass&) = delete;
+    SimpleEventClass(SimpleEventClass&&) = delete;
+
+    SimpleEventClass& operator=(const SimpleEventClass&) = delete;
+    SimpleEventClass& operator=(SimpleEventClass&&) = delete;
+
+    ~SimpleEventClass()
+    {
+        m_handleStoepsel.reset();
+        m_handleHypnotoad.reset();
+    }
+
+    void enableEvent(iox::popo::TriggerHandle&& handle, const SimpleEvent event) noexcept
+    {
+        switch (event)
+        {
+        case SimpleEvent::StoepselBachelorParty:
+            m_handleStoepsel = std::move(handle);
+            break;
+        case SimpleEvent::Hypnotoad:
+            m_handleHypnotoad = std::move(handle);
+            break;
+        }
+    }
+
+    void enableEvent(iox::popo::TriggerHandle&& handle) noexcept
+    {
+        m_handleNoEventEnum = std::move(handle);
+    }
+
+    void invalidateTrigger(const uint64_t id) noexcept
+    {
+        g_invalidateTriggerId = id;
+        if (m_handleHypnotoad.getUniqueId() == id)
+        {
+            m_handleHypnotoad.invalidate();
+        }
+        else if (m_handleStoepsel.getUniqueId() == id)
+        {
+            m_handleStoepsel.invalidate();
+        }
+        else if (m_handleNoEventEnum.getUniqueId() == id)
+        {
+            m_handleNoEventEnum.invalidate();
+        }
+    }
+
+    void disableEvent(const SimpleEvent event) noexcept
+    {
+        switch (event)
+        {
+        case SimpleEvent::StoepselBachelorParty:
+            m_handleStoepsel.reset();
+            break;
+        case SimpleEvent::Hypnotoad:
+            m_handleHypnotoad.reset();
+            break;
+        }
+    }
+
+    void disableEvent() noexcept
+    {
+        m_handleNoEventEnum.reset();
+    }
+    void triggerStoepsel() noexcept
+    {
+        m_hasTriggered.store(true);
+        m_handleStoepsel.trigger();
+    }
+
+    void resetTrigger() noexcept
```

Comment:
It seems that ``resetTrigger`` is never used, so it can probably be removed. If so, the same goes for ``m_hasTriggered``.